### PR TITLE
[WIP] Update nginx ingress helm chart

### PR DESCRIPTION
**Pull request by sladyn98**
#### Is this a new chart
- [ ] `CODEOWNERS` is updated
- [ ] If secrets are needed, `.sops.yaml` must be updated with appropriated public gpg key ID, and full public gpg key must be provided to @olblak

#### What this PR does / why we need it:
If we don't want any downtime, we'll have to deploy a new ingress controller with a new public IP then switch each dns record to the new IP one by one. 

Step:1 Add a new helm traefik helm chart

I would like to take the opportunity to switch to Traefik

#### Which issue this PR fixes

  - fixes # [INFRA-2743](https://issues.jenkins-ci.org/browse/INFRA-2743)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by sladyn98** [comment]
This is my first time setting up a helm chart would love some help @olblak @halkeye

**Review by markyjackson-taulia — CHANGES_REQUESTED** [review]
I do not see any changes

**Comment by sladyn98** [comment]
Sorry :( wow

**Review by markyjackson-taulia — CHANGES_REQUESTED** [review]
Have you run a `helm verify` and `helm lint`?
The version looks off and I am not sure we want this in the `kube-system` namespace

File: helmfile.d/traefik.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+repositories:
+  - name: traefik
+    url:
```
Comment (halkeye) [inline helmfile.d/traefik.yaml:3]: There's no url for the traefik repo, im not sure what it should be but it needs to be something

File: helmfile.d/traefik.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+releases:
+- name: traefik
+  chart: stable/traefik
+  namespace: kube-system
+  version: 1.3.0
+  wait: true
+  timeout: 300
+  atomic: true
+values:
+  - defaultBackend:
```
Comment (halkeye) [inline helmfile.d/traefik.yaml:13]: Normally we use value files instead of defining them directly in the helmfile

File: helmfile.d/traefik.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      enabled: true
+      image:
+        repository: jenkinsciinfra/404
```
Comment (halkeye) [inline helmfile.d/traefik.yaml:16]: This is just to replace the default 404 with something more jenkensie?

File: helmfile.d/traefik.yaml (hunk 2 above)
Comment (halkeye) [reply helmfile.d/traefik.yaml:13]: The only traefik helmchart I could find doesn't even mention default backend so I don't think this even does anything.

File: helmfile.d/traefik.yaml (hunk 3 above)
Comment (olblak) [reply helmfile.d/traefik.yaml:16]: Yes it's just a Jenkins 404 page :-p

File: helmfile.d/traefik.yaml (hunk 2 above)
Comment (sladyn98) [reply helmfile.d/traefik.yaml:13]: https://github.com/traefik/traefik-helm-chart

Is there any reference I can pick values from. And should there be a different values file ?

**Comment by olblak** [comment]
We just did a session with @sladyn98 to test this PR together, we identified few improvements that he should fix himself and we'll probably do another session next week to finalize this PR

**Comment by olblak** [comment]
```
grafana, grafana, StatefulSet (apps) has changed:
  # Source: grafana/templates/statefulset.yaml
  apiVersion: apps/v1
  kind: StatefulSet
  metadata:
    name: grafana
    namespace: grafana
    labels:
      helm.sh/chart: grafana-6.1.16
      app.kubernetes.io/name: grafana
      app.kubernetes.io/instance: grafana
      app.kubernetes.io/version: "7.3.5"
      app.kubernetes.io/managed-by: Helm
  spec:
    replicas: 1
    selector:
      matchLabels:
        app.kubernetes.io/name: grafana
        app.kubernetes.io/instance: grafana
    serviceName: grafana-headless
    template:
      metadata:
        labels:
          app.kubernetes.io/name: grafana
          app.kubernetes.io/instance: grafana
        annotations:
          checksum/config: 39cee3da168106b8b5a3c50f1952687357f8d34661dffa4cd94cc76f52491ca9
          checksum/dashboards-json-config: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b
          checksum/sc-dashboard-provider-config: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b
-         checksum/secret: 43070c19d0ec7b3b799d0671355eb340582863725ecc1e7fe6716c1148d06514
+         checksum/secret: 0ef769b70105734e894ecaf5a745a8c61fe697ea5603f9e2b72c7b9df09d84c8
      spec:
        
        serviceAccountName: grafana
        securityContext:
          fsGroup: 472
          runAsGroup: 472
          runAsUser: 472
        initContainers:
          - name: init-chown-data
            image: "busybox:1.31.1"
            imagePullPolicy: IfNotPresent
            securityContext:
              runAsNonRoot: false
              runAsUser: 0
            command: ["chown", "-R", "472:472", "/var/lib/grafana"]
            resources:
              {}
            volumeMounts:
              - name: storage
                mountPath: "/var/lib/grafana"
        containers:
          - name: grafana
            image: "grafana/grafana:7.3.5"
            imagePullPolicy: IfNotPresent
            volumeMounts:
              - name: config
                mountPath: "/etc/grafana/grafana.ini"
                subPath: grafana.ini
              - name: ldap
                mountPath: "/etc/grafana/ldap.toml"
                subPath: ldap.toml
              - name: storage
                mountPath: "/var/lib/grafana"
              - name: config
                mountPath: "/etc/grafana/provisioning/datasources/datasources.yaml"
                subPath: datasources.yaml
            ports:
              - name: service
                containerPort: 80
                protocol: TCP
              - name: grafana
                containerPort: 3000
                protocol: TCP
            env:
              - name: GF_SECURITY_ADMIN_USER
                valueFrom:
                  secretKeyRef:
                    name: grafana
                    key: admin-user
              - name: GF_SECURITY_ADMIN_PASSWORD
                valueFrom:
                  secretKeyRef:
                    name: grafana
                    key: admin-password
              
            livenessProbe:
              failureThreshold: 10
              httpGet:
                path: /api/health
                port: 3000
              initialDelaySeconds: 60
              timeoutSeconds: 30
            readinessProbe:
              httpGet:
                path: /api/health
                port: 3000
            resources:
              limits:
                cpu: 200m
                memory: 256Mi
              requests:
                cpu: 100m
                memory: 128Mi
        volumes:
          - name: config
            configMap:
              name: grafana
          - name: ldap
            secret:
              secretName: grafana
              items:
                - key: ldap-toml
                  path: ldap.toml
        # nothing
    volumeClaimTemplates:
    - metadata:
        name: storage
      spec:
        accessModes: [ReadWriteOnce]
        storageClassName: 
        resources:
          requests:
            storage: 50
grafana, grafana, Secret (v1) has changed:
+ Changes suppressed on sensitive content of type Secret

jenkins-infra, jenkins-infra, Secret (v1) has changed:
+ Changes suppressed on sensitive content of type Secret

release, default-release-jenkins, Secret (v1) has changed:
+ Changes suppressed on sensitive content of type Secret

**Pull request closed without merging**